forte/data/ontology/core.py

```diff
@@ -221,6 +221,11 @@ def _check_attr_type(self, key, value):
         """
         if key not in default_entry_fields:
             hints = get_type_hints(self.__class__)
+            if key not in hints.keys():
+                raise KeyError(
```

Comment:
I actually suggest converting this to a warning. I realize that a type hint may not necessarily present. Could you also change the line below `TypeError` to a warning as well? Here, both the warnings can be "based on type annotation, blah blah."

There will be runtime exceptions being thrown when the insertion is uncessful. Btw, can you add test cases for these, to show that warnings and exceptions are thrown as expected?